specification/containerservice/Fleet.Management/update/autoupgradeprofile.tsp

```diff
@@ -171,8 +171,8 @@ interface AutoUpgradeProfileOperations {
     void,
     GenerateResponse,
     BaseParameters<AutoUpgradeProfile>,
-    LroHeaders = ArmAsyncOperationHeader &
-      ArmLroLocationHeader &
+    LroHeaders = ArmAsyncOperationHeader<FinalResult = GenerateResponse> &
```

Comment:
can you provide a link to the issue in the JS or Typescript generation that forces you to set these explicitely, as opposed to the other languages?

Reply:
my understanding is this is a cross-language issue not sure this will not impact other langauges, @weidongxu-microsoft @msyyc @tadelesh i need your expertise. could you confirm other languages?

Reply:
Offlined confirmed internally, Java has some workaround for this so their return type is correct, Python is generating from swagger this time so SDK is correct. And other languages like JS and Go would benefit for this fix.

Reply:
@MaryGao How does js handle `ArmAsyncOperationHeader<FinalResult = GenerateResponse>`?
`Azure-AsyncOperation` requires response body be of a [special format](https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#azure-asyncoperation-resource-format):
![image](https://github.com/user-attachments/assets/6e54584a-9bf9-46ac-b097-475802dd80f0)

Currently Java relies solely on `Location` header for POST final result, where the response body schema is the actual FinalResult.

Reply:
@serbrech 

If you check the template of `ArmAsyncOperationHeader`
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-azure-resource-manager/lib/operations.tsp#L287-L298

You can see as default, the `FinalResult` on `LroHeaders` is always the `Response` of the template variable, which is the `GenerateResponse` model.

But when one does not use that default (as here), one would had to put that `Response = GenerateResponse` into this `FinalResult`

Reply:
thanks a lot for the clarification, that makes sense